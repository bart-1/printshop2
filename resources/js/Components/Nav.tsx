import React, { FC } from "react";
import Button from "./Button";
import NavLink from "./NavLink";

import { RouteProps } from "../Layouts/Template";

const Nav: FC<RouteProps> = ({ navRoutes, auth }) => {
    // let adminNav;
    // if (auth && auth.user && auth.user.admin === 1) {
    //     adminNav = navRoutes;
    // } else {
    //     adminNav = navRoutes?.filter((route) => route !== "admin-tools");
    // }
    const navButtonFactory = navRoutes?.map((route, index) => (
        <NavLink key={index} href={"/" + route} active={true}>
            <Button
                type="button"
                onClick={() => null}
                className="mr-3"
                disabled={false}
                size="m"
            >
                {route}
            </Button>
        </NavLink>
    ));

    return <div className="mt-6 mb-6">{navButtonFactory}</div>;
};

export default Nav;
