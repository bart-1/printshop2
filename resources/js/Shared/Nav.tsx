import React, { FC } from "react";
import { ComponentsRouteProps } from "../Layouts/Template";
import NavLinkButton from "./NavLinkButton";

interface NavProps extends ComponentsRouteProps {
    classNameBtn: string;
    classNameDiv: string;
}

const Nav = ({ navRoutes, classNameBtn, classNameDiv }: NavProps) => {
    const navButtonFactory = navRoutes?.map((route, index) => (
        <NavLinkButton
            key={index}
            href={"/" + route}
            active={true}
            className={`${classNameBtn}`}
            disabled={false}
            size="s"
            dataCy={route}
        >
            {route}
        </NavLinkButton>
    ));

    return <div className={classNameDiv}>{navButtonFactory}</div>;
};

export default Nav;
