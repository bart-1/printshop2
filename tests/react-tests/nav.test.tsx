import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import renderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";

import Nav from "../../resources/js/Shared/Nav";
import { findDOMNode } from "react-dom";

const navRoutes = ["route-one", "route-two"];

test("buttons are created based strings array in props", () => {
    const { getByText, queryByText } = render(<Nav navRoutes={navRoutes} />);
    navRoutes.map((route) => {
        expect(queryByText(route)).toBeTruthy;
        fireEvent.click(screen.getByText(route));
    });
});

test("all generated btns have proper route value in a tag href attr", () => {
    const { container } = render(<Nav navRoutes={navRoutes} />);
    const a = container.querySelectorAll("a");
    navRoutes.map((route, index) => {
        const href = a[index].getAttribute("href");
        expect(href).toBe(`/${route}`);
    });
});
