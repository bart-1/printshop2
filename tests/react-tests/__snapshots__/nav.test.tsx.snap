// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all generated buttons have proper href value 1`] = `
<div
  className="mt-6 mb-6"
>
  <a
    className=""
    href="/route-one"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-one
    </button>
  </a>
  <a
    className=""
    href="/route-two"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-two
    </button>
  </a>
</div>
`;

exports[`all generated buttons have proper href value 2`] = `
<div
  className="mt-6 mb-6"
>
  <a
    className=""
    href="/route-one"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-one
    </button>
  </a>
  <a
    className=""
    href="/route-two"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-two
    </button>
  </a>
</div>
`;

exports[`all generated buttons have proper href value 3`] = `
<div
  className="mt-6 mb-6"
>
  <a
    className=""
    href="/route-one"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-one
    </button>
  </a>
  <a
    className=""
    href="/route-two"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-two
    </button>
  </a>
</div>
`;

exports[`all generated buttons have proper href value 4`] = `
<div
  className="mt-6 mb-6"
>
  <a
    className=""
    href="/route-one"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-one
    </button>
  </a>
  <a
    className=""
    href="/route-two"
    onClick={[Function]}
  >
    <button
      className="inline-flex items-center border-transparent rounded-md font-semibold uppercase tracking-widest false px-4 py-2 border-2 text-xs false  text-[color:var(--my-background)] hover:bg-[color:var(--my-background)]   bg-[color:var(--my-foreground)] hover:border-[color:var(--my-foreground)] hover:text-[color:var(--my-foreground)]     false mr-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      route-two
    </button>
  </a>
</div>
`;
